server/datastore/mysql/hosts.go

```diff
@@ -733,6 +733,56 @@ func shouldCleanTeamPolicies(currentTeamID, newTeamID *uint) bool {
 	return *currentTeamID != *newTeamID
 }
 
+func (ds *Datastore) EnrollOrbit(ctx context.Context, hardwareUUID string, orbitNodeKey string) (*fleet.Host, error) {
+	if orbitNodeKey == "" {
+		return nil, ctxerr.New(ctx, "orbit node key is empty")
+	}
+
+	if hardwareUUID == "" {
+		return nil, ctxerr.New(ctx, "harware uuid is empty")
+	}
+
+	var host fleet.Host
+	err := ds.withRetryTxx(ctx, func(tx sqlx.ExtContext) error {
+		err := sqlx.GetContext(ctx, tx, &host, `SELECT id FROM hosts WHERE uuid = ?`, hardwareUUID)
+		switch {
+		case err == nil:
+			sqlUpdate := `UPDATE hosts SET orbit_node_key = ? WHERE uuid = ? `
+			_, err := tx.ExecContext(ctx, sqlUpdate, orbitNodeKey, hardwareUUID)
+			if err != nil {
+				return ctxerr.Wrap(ctx, err, "orbit enroll error updating host details")
+			}
+		case errors.Is(err, sql.ErrNoRows):
+			zeroTime := time.Unix(0, 0).Add(24 * time.Hour)
+			// create a new host
+			sqlInsert := `
+				INSERT INTO hosts (
+					last_enrolled_at,               
+					detail_updated_at,
+					label_updated_at,
+					policy_updated_at,
+					osquery_host_id,
+					node_key,
```

Comment:
Problem with sertting `node_key` to `""` is that this column is unique so if two Orbit instances are enrolling at the same time it may fail. Is there an issue with storing it as `NULL`? (Plus there's the security issue of allowing authentication with empty node key)

Reply:
Can we set it as `NULL`, since `Host.NodeKey` is `string` and not a pointer? I think `NULL` should be okay, since `hosts` table schema says default can be null...

But yeah, all ears if you have suggestions on how to handle it

Reply:
Ah, the other (safe) option is to set it with same value as `orbit_node_key`.

Reply:
Sounds good -- I think that works best, updating